tests/run_test.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/sh
+
+usage() {
+  cat >&2 << HELP
```

Comment:
consider using quotes with the heredoc delimiter

Reply:
I don't follow. This is the same pattern I used in all the other tools scripts.

Reply:
I meant cat << 'HELP'... This minor difference avoids the unnecessary scan for embedded $xxx variables.

Reply:
So the heredoc delimiter itself contains quotes? That seems confusing. I think I could also use a single quoted expression here, which would avoid the need for heredoc.

Reply:
`man bash` on my Ubuntu system says:

> If any part of word is quoted,  the  delimiter is  the  result of quote removal on word, and the lines in the here-document are not expanded.  If word is unquoted, all lines of the here-document are subjected to  parameter expansion, command substitution, and arithmetic expansion, the character sequence \<new‐line> is ignored, and \ must be used to quote the characters \, $, and `.

There is similar text in `man dash` so I suppose this is likely to be portable to other shells. It would appear to solve the problem you describe with backticks.

Reply:
Oh, I didn't realize that was a special syntax. I still think it's weird, but if that's the standard thing then okay.